#!/usr/bin/env python
PACKAGE = "peripheralsInterface"

import roslib;
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("motorPlotEnabled", bool_t, 0, "Enable Data Dump for plotting", False)
gen.add("Kp", int_t, 0, "P-value of PID", 2560, 0, 65536)
gen.add("Ki", int_t, 0, "I-value of PID", 768, 0, 65536)
gen.add("Kd", int_t, 0, "D-value of PID", 0, 0, 65536)
gen.add("iTh", int_t, 0, "Threshold of using I (iMin)", 512, 0, 65536)
gen.add("iMax", int_t, 0, "Limits the maximum I value", 4000, 0, 100000)
gen.add("PwmMax", int_t, 0, "PWM Maximum power (max=648)", 648, 0, 648)
gen.add("PwmMaxDeltaSize", int_t, 0, "PWM maximum step size per cycle (2.5ms, 400Hz, max=648<<16)", 65536, 0, 10*65536)

gen.add("X1", double_t, 0, "The X value for the left motor", -0.5, -10.0, 10.0)
gen.add("X2", double_t, 0, "The X value for the right motor", -0.5, -10.0, 10.0)
gen.add("X3", double_t, 0, "The X value for the rear motor", 1.0, -10.0, 10.0)

gen.add("Y1", double_t, 0, "The Y value for the left motor", -0.866, -10.0, 10.0)
gen.add("Y2", double_t, 0, "The Y value for the right motor", 0.866, -10.0, 10.0)
gen.add("Y3", double_t, 0, "The Y value for the rear motor", 0.0, -10.0, 10.0)

gen.add("Theta1", double_t, 0, "The Theta value for the left motor", 0.22, -10.0, 10.0)
gen.add("Theta2", double_t, 0, "The Theta value for the right motor", 0.22, -10.0, 10.0)
gen.add("Theta3", double_t, 0, "The Theta value for the rear motor", 0.2266, -10.0, 10.0)

exit(gen.generate(PACKAGE, "Motors", "motors"))
